integration/sandbox_clean_remove_test.go

```diff
@@ -124,3 +124,108 @@ func TestSandboxRemoveWithoutIPLeakage(t *testing.T) {
 	t.Logf("Should not be able to find the pod ip in host-local checkpoint")
 	assert.False(t, checkIP(ip))
 }
+
+// Issue: https://github.com/containerd/containerd/issues/5768
+// PR: https://github.com/containerd/containerd/pull/5904
+// This test simulates the network setup failure and teardown failure by renaming the name of portmap binary.
+// Due to the nature of chained cni plugin, host-local ipam should have allocated IP address.
+// RunPodSandx should leave such sandbox in the metadata store to cleanup IP address later.
+func TestNetworkTeardownFailureWithoutIPLeakage(t *testing.T) {
+
+	t.Logf("Make sure host-local ipam and portmap are in use")
+	config, err := CRIConfig()
+	require.NoError(t, err)
+	fs, err := os.ReadDir(config.NetworkPluginConfDir)
+	require.NoError(t, err)
+	require.NotEmpty(t, fs)
+	f := filepath.Join(config.NetworkPluginConfDir, fs[0].Name())
+	cniConfig, err := os.ReadFile(f)
+	require.NoError(t, err)
+	if !strings.Contains(string(cniConfig), "host-local") {
+		t.Skip("host-local ipam is not in use")
+	}
+	if !strings.Contains(string(cniConfig), "portmap") {
+		t.Skip("portmap is not in use")
+	}
+
+	t.Logf("Rename the portmap binary")
+	portmapBin := filepath.Join(config.NetworkPluginBinDir, "portmap")
+	err = os.Rename(portmapBin, portmapBin+"_tmp")
+	require.NoError(t, err)
```

Comment:
It'd be worthwhile to leverage [`t.Cleanup`](https://pkg.go.dev/testing#T.Cleanup) here and do so individually for each resource rather than having a single large `defer`.  This helps keep the cleanup local to the callsite where the resource is modified so future code changes are less likely to break the functioning of the cleanup.

```suggestion
	t.Logf("Rename the portmap binary")
	portmapBin := filepath.Join(config.NetworkPluginBinDir, "portmap")
	err = os.Rename(portmapBin, portmapBin+"_tmp")
	require.NoError(t, err)
	t.Cleanup(func() {
		os.Rename(portmapBin+"_tmp", portmapBin)
	})
```